students/Cracolici_Jon/lesson06/test_JonCracolici_Mailroom4.py

```diff
@@ -0,0 +1,131 @@
+        db2 = mr.new_donor(new_donor_dict, db)
+        william_gates_iii = {'f_name': 'William_Gates_III', 'donor': 'William Gates, III', 'total_donated': 653784.49,
+                         'num_donations': 2, 'avg_donated': 326892.24, 'last_donation': 30000.00}
+        mark_zuckerberg = {'f_name': 'Mark_Zuckerberg', 'donor': 'Mark Zuckerberg', 'total_donated': 16396.10,
+                       'num_donations': 3, 'avg_donated': 5465.37, 'last_donation': 5000.25}
+        jeff_bezos = {'f_name': 'Jeff_Bezos', 'donor': 'Jeff Bezos', 'total_donated': 877.33, 'num_donations': 1,
+                  'avg_donated': 877.33, 'last_donation': 877.33}
+        paul_allen = {'f_name': 'Paul_Allen', 'donor': 'Paul Allen', 'total_donated': 708.42, 'num_donations': 3,
+                  'avg_donated': 236.14, 'last_donation': 100.95}
+        van_halen = {'f_name': 'Van_Halen', 'donor': 'Van Halen', 'total_donated': 1000.00, 'num_donations': 1,
+                  'avg_donated': 1000.00, 'last_donation': 1000.00}
+
+        db3 = {'william_gates_iii': william_gates_iii, 'mark_zuckerberg': mark_zuckerberg, 'jeff_bezos': jeff_bezos,
+          'paul_allen': paul_allen, 'van_halen': van_halen}
+        self.assertTrue(db2, db3)
+
+    def test_note_gen(self):
+        william_gates_iii = {'f_name': 'William_Gates_III', 'donor': 'William Gates, III', 'total_donated': 653784.49,
+                         'num_donations': 2, 'avg_donated': 326892.24, 'last_donation': 30000.00}
+        a = mr.note_gen(william_gates_iii, dest='s')
+        b = mr.note_gen(william_gates_iii, dest='f')
+        message = "Dear William Gates, III, \nThank you for your generous donation of $30000.00. " \
+                  "Please rest assured that we will use at least \n95% of your contribution to feed the homeless" \
+                  " to wolves. We could not do this work without you. \nSincerely, \nThe Billionaires' Club"
+        self.assertEqual(a, print(message))
+        self.assertEqual(b, message)
+
+    def test_name_list(self):
+        db = mr.initialize_database()
+        a = mr.show_current_names(db)
+        b = 'The current donors are William Gates, III, Mark Zuckerberg, Jeff Bezos, and Paul Allen.'
+        self.assertEqual(a, print(b))
+
+    def test_report(self):
+        db = mr.initialize_database()
+        a = mr.create_report(db)
+        b = 'Donor Name                | Total Given | Num Gifts | Average Gift' \
+            '\n------------------------------------------------------------------'\
+            '\nWilliam Gates, III        $    653784.49           2 $   326892.24'\
+            '\nMark Zuckerberg           $     16396.10           3 $     5465.37'\
+            '\nJeff Bezos                $       877.33           1 $      877.33'\
+            '\nPaul Allen                $       708.42           3 $      236.14'
+
+        self.assertEqual(a, print(b))
+
+
+if __name__ == '__main__':
+    unittest.main()
```

Comment:
I would highly recommend to be intentional about naming your functions and variable names.

It makes it difficult to read the code and especially tests when the reader sees things like `a` and `b`; it is also difficult to understand what the code function being tested does. It is very common for other devs to look at unit tests to understand better how a particular function behaves, so having high quality explicit tests is a must.